src/test/java/com/example/democamel/route/MyRoutesTest.java

```diff
@@ -0,0 +1,151 @@
+package com.example.democamel.route;
+
+import com.example.democamel.model.OrderEntity;
+import com.example.democamel.model.OrderFromCsv;
+import com.example.democamel.model.RegionReport;
+import com.example.democamel.repo.OrdersOnlineRepository;
+import com.example.democamel.repo.RegionReportRepository;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.spring.junit5.CamelSpringBootTest;
+import org.apache.camel.test.spring.junit5.MockEndpoints;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+
+
+@SpringBootTest
+@CamelSpringBootTest
+@MockEndpoints
+class MyRoutesTest {
+    @Autowired
+    private ProducerTemplate template;
+
+    @EndpointInject("mock:direct:onlineOrdersToDb")
+    private MockEndpoint mock;
+
+    @EndpointInject("mock:file:output/reports?fileName=${header.region}_${date:now:yyyy-MM-dd HH-mm-ss}")
+    private MockEndpoint mockToCsv;
+
+    @Autowired
+    OrdersOnlineRepository onlineRepository;
+
+    @Autowired
+    RegionReportRepository reportRepository;
+
+    @AfterEach
+    void clearRepo() {
+        onlineRepository.deleteAll();
+        reportRepository.deleteAll();
+
+    }
+
+    @Test
+    void shouldUnmarshalCorrectClassObjectFromFile() {
```

Comment:
seems like this test doesn't assert
if we change `isInstanceOf(OrderFromCsv.class)` to any other class, e.g.`isInstanceOf(String.class)`, it will still pass